File: Regression/Decision-Tree-Regression.R
# Decision Tree Regression Model

# importing dataset
dataset = read.csv('Position_Salaries.csv')
dataset = dataset[2:3]

# splitting into test and training sets not required here because dataset is small
# feature scaling not required as Decision Tree Regression is not based on Euclidean Distance

#fitting Decision Tree Regression Model to dataset
#install.packages('rpart')
library(rpart)
set.seed(1234)

regressor = rpart(formula = Salary ~ .,
                  data = dataset,
                  control = rpart.control(minsplit = 1))

# predicting new values
y_pred = predict(regressor, data.frame(Level = 6.5))

# visualizing Decision Tree results
#library(ggplot2)
#ggplot() +
 # geom_point(aes(x = dataset$Level, y= dataset$Salary), color = 'red') +
#  geom_line(aes(x = dataset$Level, y = predict(regressor, newdata = dataset)), color = 'blue') +
 # ggtitle('Level vs Salary') +
  #xlab('Level') +
  #ylab('Salary')

# visualising Decision Tree results for smooth curve and high resolution
library(ggplot2)

X_grid = seq(min(dataset$Level),max(dataset$Level),0.001)
ggplot() +
  geom_point(aes(x=dataset$Level,y=dataset$Salary), color='red')+
  geom_line(aes(x=X_grid,y=predict(regressor,newdata = data.frame(Level = X_grid))), color='blue')+
  ggtitle('Level vs Salary')+
  xlab('Level')+
  ylab('Salary')

File: Regression/Decision-Tree-Regression.py
# -*- coding: utf-8 -*-
"""
Created on Mon Oct 30 15:03:25 2017

@author: adhingra
"""
# Decision Tree Regression

import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

# importing dataset
dataset = pd.read_csv('Position_Salaries.csv')
X = dataset.iloc[:,1:2].values
y = dataset.iloc[:,2].values

# splitting into test and training sets not reqd because of small dataset
# feature scaling not reqd because its not based on Euclidian Distance

# fitting dataset into Decision Tree Regression Model
from sklearn.tree import DecisionTreeRegressor
regressor = DecisionTreeRegressor(random_state = 0)
regressor.fit(X,y)

# predicting values for unknown x values
y_pred = regressor.predict(6.5)

# visualising Decision Tree Regression Results **provide continuous result**
#plt.scatter (X,y,color='red')
#plt.plot(X,regressor.predict(X), color='blue')
#plt.xlabel('Level')
#plt.ylabel('Salary')
#plt.title('Level vs Salary')
#plt.show()

# visualizing Decision Tree Regression Results (for Smoother curves and higher resolution) 
X_grid = np.arange(min(X),max(X),0.01)
X_grid = X_grid.reshape((len(X_grid),1))
plt.scatter(X,y,color='red')
plt.plot(X_grid,regressor.predict(X_grid),color='blue')
plt.xlabel('Level')
plt.ylabel('Salary')
plt.title('Level vs Salary')
plt.show()
